### PR TITLE
FIX-#1464: product/sum incorrect behavior of 'min_count' fixed

### LINKED ISSUE #1464: Need to test DataFrame.product with small DataFrames
### System information
- **OS Platform and Distribution (e.g., Linux Ubuntu 16.04)**:

Ubuntu 19.10

- **Modin version** (`modin.__version__`):

0.7.3+42.gc78828e.dirty

- **Python version**:

Python 3.7.5

- **Code we can use to reproduce**:

```patch
diff --git a/modin/pandas/test/utils.py b/modin/pandas/test/utils.py
index 1faaf38..d4d257c 100644
--- a/modin/pandas/test/utils.py
+++ b/modin/pandas/test/utils.py
@@ -63,6 +63,11 @@ test_data = {
         ]
         for i in range(NCOLS)
     },
+    "small": {
+        "col0": [1, 2, 3, 4],
+        "col1": [8.0, 9.4, 10.1, 11.3],
+        "col2": [4, 5, 6, 7],
+    },
     # "int_float_object_data": {
     #     "col3": [1, 2, 3, 4],
     #     "col4": [4, 5, 6, 7],
```

### Describe the problem

When dataframe size is small and is smaller than `min_count` parameter value for `product` it causes a lot of tests to fail. Pandas behaves quite strangely and Modin works differently.

### PR DESCRIPTION
**Pull request by dchigarev**
Signed-off-by: Dmitry Chigarev <dmitry.chigarev@intel.com>

## What do these changes do?

- [x] commit message follows format outlined [here](https://modin.readthedocs.io/en/latest/contributing.html)
- [x] passes `flake8 modin`
- [x] passes `black --check modin`
- [x] signed commit with `git commit -s` 
- [x] Resolves #1464 
- [x] tests added and passing

### CODE REVIEW

File: modin/pandas/test/test_series.py [hunk 1]
```diff
@@ -2086,19 +2088,37 @@ def test_pow(data):
...
+        ),
+    ],
+)
+@pytest.mark.parametrize(
+    "min_count", int_arg_values, ids=arg_keys("min_count", int_arg_keys)
+)
+def test_prod(data, axis, skipna, numeric_only, min_count):
+    eval_general(
+        *create_test_series(data),
+        lambda df, *args, **kwargs: type(df)([df.prod(*args, **kwargs)]),
```
Comment (anmyachev) [inline modin/pandas/test/test_series.py:2117]: You should test `product` also.

File: modin/pandas/test/test_series.py [hunk 2]
```diff
@@ -2086,19 +2088,37 @@ def test_pow(data):
...
+)
+@pytest.mark.parametrize(
+    "numeric_only",
+    [
+        None,
+        False,
+        pytest.param(
+            True,
+            marks=pytest.mark.xfail(reason="numeric_only not implemented for Series"),
+        ),
```
Comment (anmyachev) [inline modin/pandas/test/test_series.py:2108]: Why `xfail`? Is modin behavior different from pandas behavior?

File: modin/pandas/test/test_series.py [hunk 3]
```diff
@@ -2685,23 +2705,38 @@ def test_subtract(data):
...
 )
+@pytest.mark.parametrize(
+    "numeric_only",
+    [
+        None,
+        False,
+        pytest.param(
+            True,
+            marks=pytest.mark.xfail(reason="numeric_only not implemented for Series"),
+        ),
```
Comment (anmyachev) [inline modin/pandas/test/test_series.py:2725]: Same about `xfail`.

File: modin/pandas/test/test_dataframe.py [hunk 4]
```diff
@@ -3456,29 +3407,26 @@ def test_prod(self, request, data, axis, skipna, numeric_only, min_count):
...
-                    numeric_only=numeric_only,
-                    min_count=min_count,
-                )
-        else:
-            modin_result = modin_df.product(
-                axis=axis, skipna=skipna, numeric_only=numeric_only, min_count=min_count
-            )
-            df_equals(modin_result, pandas_result)
+    @pytest.mark.parametrize("is_transposed", [False, True])
+    def test_prod(
```
Comment (anmyachev) [inline modin/pandas/test/test_dataframe.py:3411]: We need to test `product` also.

File: modin/pandas/test/test_dataframe.py [hunk 5]
```diff
@@ -3456,29 +3407,26 @@ def test_prod(self, request, data, axis, skipna, numeric_only, min_count):
...
-                    skipna=skipna,
-                    numeric_only=numeric_only,
-                    min_count=min_count,
-                )
-        else:
-            modin_result = modin_df.product(
-                axis=axis, skipna=skipna, numeric_only=numeric_only, min_count=min_count
-            )
-            df_equals(modin_result, pandas_result)
+    @pytest.mark.parametrize("is_transposed", [False, True])
```
Comment (anmyachev) [inline modin/pandas/test/test_dataframe.py:3410]: Why you add the option?

File: modin/pandas/test/test_series.py (hunk 1 above)
Comment (dchigarev) [reply modin/pandas/test/test_series.py:2117]: `product` is just an alias for `prod`, I don't think we want to run 2000+ extra tests to test the same things

File: modin/pandas/test/test_series.py (hunk 2 above)
Comment (dchigarev) [reply modin/pandas/test/test_series.py:2108]: `numeric_only=True` is not implemented for `pandas.Series` and throws exception, but `modin.Series` supports this parameter, should we copy such type of behavior?

File: modin/pandas/test/test_series.py (hunk 3 above)
Comment (dchigarev) [reply modin/pandas/test/test_series.py:2725]: answered above

File: modin/pandas/test/test_dataframe.py (hunk 4 above)
Comment (dchigarev) [reply modin/pandas/test/test_dataframe.py:3411]: answered above

File: modin/pandas/test/test_dataframe.py (hunk 5 above)
Comment (dchigarev) [reply modin/pandas/test/test_dataframe.py:3410]: Previous test implementation computed `prod/sum` for transposed and not transposed data, I just moved that into test parameter

File: modin/pandas/test/test_series.py (hunk 1 above)
Comment (anmyachev) [reply modin/pandas/test/test_series.py:2117]: Is there a chance that this behavior could change in the future?

File: modin/pandas/test/test_series.py (hunk 2 above)
Comment (anmyachev) [reply modin/pandas/test/test_series.py:2108]:
```suggestion
        pytest.param(
            True,
            marks=pytest.mark.xfail(reason="numeric_only not implemented for pandas.Series"),
        ),
```

Comment (anmyachev) [reply modin/pandas/test/test_series.py:2108]: No, but let's reflect that the problem is not on our side.

File: modin/pandas/test/test_series.py (hunk 3 above)
Comment (anmyachev) [reply modin/pandas/test/test_series.py:2725]:
```suggestion
        pytest.param(
            True,
            marks=pytest.mark.xfail(reason="numeric_only not implemented for pandas.Series"),
        ),
```

File: modin/pandas/test/test_series.py (hunk 1 above)
Comment (dchigarev) [reply modin/pandas/test/test_series.py:2117]: well, there is always a chance of everything...

but for now, modin and pandas have that line in their `DataFrame`/`Series` classes:
```python
product = prod
```

Comment (anmyachev) [reply modin/pandas/test/test_series.py:2117]: Well, but we can check that `product` is alias of `prod`. When and if it doesn't work that way, we can add tests. What about that way?

Comment (dchigarev) [reply modin/pandas/test/test_series.py]: done it

**Comment by devin-petersohn** [comment]
Will wait for @anmyachev to approve.

**Pull request merged**